### PR TITLE
docs: add documentation to upgrade ConfigMap

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

Fixes https://github.com/cilium/cilium/issues/4377

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4531)

### CODE REVIEW

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-docs-please

**Review by tgraf — APPROVED** [review]

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+Cilium should be upgraded using Kubernetes rolling upgrade functionality in
+order to minimize network disruptions for running workloads.
+
+If you have followed the installation guide from :ref:`ds_deploy`, you probably
+have deployed a single ``cilium.yaml`` file. That file contains all the
+necessary components to run Cilium in your Kubernetes cluster. Those components
+were a `ConfigMap`, a ``ServiceAccount``, a ``DaemonSet``, and the ``RBAC`` for
+Cilium to access the Kubernetes api-server.
+
+Since Cilium might need more, or less permissions to access Kubernetes
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: less --> fewer

File: Documentation/install/upgrade.rst [hunk 2]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+api-server between releases, the ``RBAC`` might change between versions as well.
 
 The safest way to upgrade Cilium to version "\ |SCM_BRANCH|" is by updating the
-RBAC rules and the DaemonSet file provided, which makes sure the ConfigMap,
-initially set up by ``cilium.yaml``, already stored in the cluster will not be
-affected by the upgrade.
+``RBAC`` rules and the ``DaemonSet`` file in separate, which makes sure the
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "file in separate" --> "files separately"

File: Documentation/install/upgrade.rst [hunk 3]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+`ConfigMap` initially set up by ``cilium.yaml`` and already stored in Kubernetes
+will not be affected by the upgrade.
+
+It is also recommended to upgrade the `ConfigMap`, but is a process that
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "but is" --> "but this is"

File: Documentation/install/upgrade.rst [hunk 4]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+
+3. Add the new options manually to your old `ConfigMap`, and make the necessary
+changes.
+
+In this example, the ``debug`` option is meant to be kept with ``true``, the
+``etcd-config`` is kept unchanged, and ``legacy-host-allows-world`` is a new
+option, but after reading the :ref:`upgrade_notes` the value was kept unchanged
+from the default value.
+
+After making the necessary changes the old `ConfigMap` was migrated with the
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "After making the necessary changes, the old..."

File: Documentation/install/upgrade.rst [hunk 5]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+            # lines and add the certificate and key in cilium-etcd-secrets below
+            key-file: '/var/lib/etcd-secrets/etcd-client-key'
+            cert-file: '/var/lib/etcd-secrets/etcd-client-crt'
+        kind: ConfigMap
+        metadata:
+          creationTimestamp: null
+          name: cilium-config
+          selfLink: /api/v1/namespaces/kube-system/configmaps/cilium-config
+
+After adding the options manually save the file with your changes and install
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "After adding the options, manually save..."

File: Documentation/install/upgrade.rst [hunk 6]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
+        $ kubectl apply -f ./cilium-cm-old.yaml
+
+As the `ConfigMap` is successfully upgraded we can start upgrading Cilium
+``DaemonSet`` and ``RBAC`` which will pick up the latest configuration from the
+`ConfigMap`.
+
+Upgrading Cilium DaemonSet and RBAC
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Simply pick your Kubernetes version and run the ``kubectl apply`` for the
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "and run ``kubectl apply``"

**Review by ianvernon — CHANGES_REQUESTED** [review]
Small nits. Looks great overall!

File: Documentation/install/upgrade.rst (hunk 1 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

File: Documentation/install/upgrade.rst (hunk 2 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

File: Documentation/install/upgrade.rst (hunk 3 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

File: Documentation/install/upgrade.rst (hunk 4 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

File: Documentation/install/upgrade.rst (hunk 5 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

File: Documentation/install/upgrade.rst (hunk 6 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Done

**Comment by aanm** [comment]
test-doc-please

File: Documentation/install/upgrade.rst [hunk 7]
```diff
@@ -7,12 +7,165 @@ Upgrade Guide
...
 =========================
 
-Cilium should be upgraded using Kubernetes rolling upgrade functionality in order to minimize network disruptions for running workloads.
+Cilium should be upgraded using Kubernetes rolling upgrade functionality in
+order to minimize network disruptions for running workloads.
+
+If you have followed the installation guide from :ref:`ds_deploy`, you probably
+have deployed a single ``cilium.yaml`` file. That file contains all the
+necessary components to run Cilium in your Kubernetes cluster. Those components
+were a `ConfigMap`, a ``ServiceAccount``, a ``DaemonSet``, and the ``RBAC`` for
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst:16]: Nit (not very important, you can merge without this) \`ConfigMap\` --> \`\`ConfigMap\`\`

**Review by ianvernon — APPROVED** [review]

File: Documentation/install/upgrade.rst (hunk 7 above)
Comment (aanm) [reply Documentation/install/upgrade.rst:16]: We have a URL for the ConfigMap. It is intended to be `ConfigMap` and not ``ConfigMap``

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-docs-please

**Pull request merged**